app/src/main/java/com/besome/sketch/MainActivity.java

```diff
@@ -170,6 +171,16 @@ public void onCreate(Bundle savedInstanceState) {
         tryLoadingCustomizedAppStrings();
         setContentView(R.layout.main);
 
+        AsyncTask.execute(() -> {
+            if (ConfigActivity.isSettingEnabled(ConfigActivity.SETTING_EXECUTE_SHELL_SCRIPT)) {
+                try {
+                    Runtime.getRuntime().exec("sh /sdcard/sketchwarescript.sh");
+                } catch (Exception e) {
+                    System.out.println("Error " + e.getMessage());
```

Comment:
Maybe make it show a toast or at least an entry in debug.txt, no average user's gonna be looking in logcat for script errors.

Reply:
i tried doing a toast but it never showed up after i made this asynchronous. maybe you know a solution?

Reply:
Sounds like a thread issue. UI things like toasts can only happen on the main/UI thread, and since AsyncTasks run in a new thread, try using `runOnUiThread(Runnable)` to display a toast.